x-pack/metricbeat/module/aws/awshealth/awshealth.go

```diff
@@ -0,0 +1,431 @@
+		pending            int32
+		resolved           int32
+		others             int32
+	)
+
+	for {
+		// When invoking the DescribeAffectedEntities for the first time, there must not exist any NextToken.
+		// DescribeAffectedEntities API call will return the next token if there are more records left for querying
+		// If there exist no further records to fetch, next toke will be empty.
+		if affEntityTokString == "" {
+			affectedEntities, err := awsHealth.DescribeAffectedEntities(ctx, &health.DescribeAffectedEntitiesInput{
+				Filter: &types.EntityFilter{
+					EventArns: []string{*event.Arn},
+				},
+				Locale:     &locale,
+				MaxResults: &maxResults,
+			})
+			if err != nil {
+				err = fmt.Errorf("AWS Health DescribeAffectedEntities failed with : %w", err)
+
+				// Check if the error is due to context cancellation
+				if errors.Is(err, context.Canceled) {
+					m.Logger().Debug("Context cancelled. Exiting gracefully.")
+					return nil
+				}
+				return err
+			}
+			if affectedEntities != nil {
+				nextToken = affectedEntities.NextToken
+
+				hd.affectedEntities = append(hd.affectedEntities, affectedEntities.Entities...)
+				for _, affEntity := range affectedEntities.Entities {
+					switch affEntity.StatusCode {
+					case "PENDING":
+						pending++
+					case "RESOLVED":
+						resolved++
+					case "":
+						// Do nothing
+					default:
+						others++
+					}
+				}
+			}
+
+		} else {
+			affectedEntities, err := awsHealth.DescribeAffectedEntities(ctx, &health.DescribeAffectedEntitiesInput{
+				Filter: &types.EntityFilter{
+					EventArns: []string{*event.Arn},
+				},
+				Locale:     &locale,
+				MaxResults: &maxResults,
+				NextToken:  &affEntityTokString,
+			})
+			if err != nil {
+				err = fmt.Errorf("AWS Health DescribeAffectedEntities failed with : %w", err)
+
+				// Check if the error is due to context cancellation
+				if errors.Is(err, context.Canceled) {
+					m.Logger().Info("Context cancelled. Exiting gracefully.")
+					return nil
+				}
+				return err
+			}
+			if affectedEntities != nil {
+				nextToken = affectedEntities.NextToken
+				hd.affectedEntities = append(hd.affectedEntities, affectedEntities.Entities...)
+
+				for _, affEntity := range affectedEntities.Entities {
+					switch affEntity.StatusCode {
+					case "PENDING":
+						pending++
+					case "RESOLVED":
+						resolved++
+					case "":
+						// Do nothing
+					default:
+						others++
+					}
+				}
+			}
+		}
+		if nextToken == nil {
+			break
+		}
+		affEntityTokString = *nextToken
+	}
+	hd.affectedEntityResolved = resolved
+	hd.affectedEntityPending = pending
+	hd.affectedEntityOthers = others
+
+	select {
+	case ch <- hd:
+		// Writing to the channel
+	default:
+		// Channel buffer is full or closed, dropping the event to avoid blocking.
+		return nil
+	}
+	return nil
+}
+
+func getCurrentDateTime() string {
+	// Reference: https://golang.org/pkg/time/#Time.Format
+	return time.Now().Format("20060102150405")
```

Comment:
what specifies this format? as in what is the reason we are using this specific time format?